builder/dockerfile/parser/parser.go

```diff
@@ -101,19 +103,24 @@ func parseLine(line string) (string, *Node, error) {
 // Parse is the main parse routine.
 // It handles an io.ReadWriteCloser and returns the root of the AST.
 func Parse(rwc io.Reader) (*Node, error) {
+	currentLine := 0
 	root := &Node{}
+	root.StartLine = currentLine
```

Comment:
Skipping the empty lines is a very good idea. I changed the code to set the startLine as the first non-empty line and the endLine as the last non-empty line.